src/fuzzycmeans.jl

```diff
@@ -0,0 +1,112 @@
+# Fuzzy C means algorithm
+
+## Interface
+
+immutable FuzzyCMeansResult{T<:AbstractFloat} <: ClusteringResult
+    centers::Matrix{T}          # cluster centers (d x C)
+    weights::Matrix{T}          # assigned weights (n x C)
+    iterations::Int             # number of elasped iterations
+    converged::Bool             # wheather the procedure converged
+end
+
+## Utility functions
+
+function update_weights!(weights, data, centers, fuzziness, dist_metric)
+    pow = 2.0/(fuzziness-1)
+    nrows, ncols = size(weights)
+    dists = pairwise(dist_metric, data, centers)
+    for i in 1:nrows
+        for j in 1:ncols
+            den = 0.0
+            for k in 1:ncols
+                den += (dists[i,j]/dists[i,k])^pow
+            end
+            weights[i,j] = 1.0/den
+        end
+    end
+end
+
+function update_centers!(centers, data, weights, fuzziness)
+    nrows, ncols = size(weights)
+    for j in 1:ncols
+        num = zeros(Float64, size(data[:,1]))
```

Comment:
`data[:,1]` makes a copy here.  Use `size(data, 1)`.

Also, if you update `centers` below, then this should use `zeros(T, size(data,1))` (after getting the eltype of centers, of course).